### DISCUSSION TITLE
v1.9.0

### DISCUSSION THREAD

**Discussion by nanne007** [starcoinorg/starcoin#3103] [Release]
## What's Changed
* [scripts] Add prometheus script and cleanup out date scripts. by @jolestar in https://github.com/starcoinorg/starcoin/pull/3001
* pprof use only linux by @nkysg in https://github.com/starcoinorg/starcoin/pull/3004
* unsave block body by @nkysg in https://github.com/starcoinorg/starcoin/pull/2995
* Bump pprof from 0.4.4 to 0.6.0 by @dependabot in https://github.com/starcoinorg/starcoin/pull/2996
* Bump pprof from 0.6.0 to 0.6.1 by @dependabot in https://github.com/starcoinorg/starcoin/pull/3006
* cachestorage  remove null value , db export add subcommand， improve chain error report by @nkysg in https://github.com/starcoinorg/starcoin/pull/3008
* fix docker image name by @nkysg in https://github.com/starcoinorg/starcoin/pull/3012
* move: fix publish multiple modules in move-cli by @lerencao in https://github.com/starcoinorg/starcoin/pull/3014
* [vm] Implement native function ecrecover by @jolestar in https://github.com/starcoinorg/starcoin/pull/3015
* export and import block range by @nkysg in https://github.com/starcoinorg/starcoin/pull/3017
* make vmconfig constants in move code by @lerencao in https://github.com/starcoinorg/starcoin/pull/3019
* Export ecrecover and ripemd160 to Move and add test by @jolestar in https://github.com/starcoinorg/starcoin/pull/3020
* fix node missing by @lerencao in https://github.com/starcoinorg/starcoin/pull/3023
* save failed block with config crate_version by @nkysg in https://github.com/starcoinorg/starcoin/pull/3022
* Bump libloading from 0.7.1 to 0.7.2 by @dependabot in https://github.com/starcoinorg/starcoin/pull/3026
* [move-unit-test] Fix move unit test cost table bug. by @jolestar in https://github.com/starcoinorg/starcoin/pull/3030
* [genesis] reset halley genesis by @jolestar in https://github.com/starcoinorg/starcoin/pull/3029
* Fix export and import block bug by @nkysg in https://github.com/starcoinorg/starcoin/pull/3031
* Bump arc-swap from 1.4.0 to 1.5.0 by @dependabot in https://github.com/starcoinorg/starcoin/pull/3025
* Fix dev block produce bug by @jolestar in https://github.com/starcoinorg/starcoin/pull/3037
* Create import_main_net_block.sh by @nkysg in https://github.com/starcoinorg/starcoin/pull/3042
* benchmark flamegraph save on s3 by @nkysg in https://github.com/starcoinorg/starcoin/pull/3043
* Native u256 by @lerencao in https://github.com/starcoinorg/starcoin/pull/3032
* Add precise logger for tx broadcast by @sanlee42 in https://github.com/starcoinorg/starcoin/pull/3048
* import barnard block and improve apply block print by @nkysg in https://github.com/starcoinorg/starcoin/pull/3054
* [accumulator] Support proof for in memory accumulator by @jolestar in https://github.com/starcoinorg/starcoin/pull/3039
* make some vector method native by @lerencao in https://github.com/starcoinorg/starcoin/pull/3055
* release stdlib v10 by @lerencao in https://github.com/starcoinorg/starcoin/pull/3059
* slog duplicated to stdout && logger user addr for rpc call by @sanlee42 in https://github.com/starcoinorg/starcoin/pull/3065
* move-cli: fix error of `missing UnitTest.move` by @lerencao in https://github.com/starcoinorg/starcoin/pull/3066
* [proof] Implement TransactionInfoWithProof by @jolestar in https://github.com/starcoinorg/starcoin/pull/3044
* jsonrpc:: logger params by @sanlee42 in https://github.com/starcoinorg/starcoin/pull/3067
* [transaction_info] Refactor transaction info and transaction info storage by @jolestar in https://github.com/starcoinorg/starcoin/pull/3061
* [merkle] Support compute root by non-existence proof and add a test by @jolestar in https://github.com/starcoinorg/starcoin/pull/3010
* optimize slog by @sanlee42 in https://github.com/starcoinorg/starcoin/pull/3070
* release v1.9.0-alpha by @lerencao in https://github.com/starcoinorg/starcoin/pull/3072
* db upgrade nonsync by @nkysg in https://github.com/starcoinorg/starcoin/pull/3076
* Bump itertools from 0.10.1 to 0.10.3 by @dependabot in https://github.com/starcoinorg/starcoin/pull/3071
* Bump sha3 from 0.9.1 to 0.10.0 by @dependabot in https://github.com/starcoinorg/starcoin/pull/3074
* kube: liveness and readiness probe by @sanlee42 in https://github.com/starcoinorg/starcoin/pull/3085
* genesis: reset proxima to start from stdlib v9 by @lerencao in https://github.com/starcoinorg/starcoin/pull/3086
* release v1.9.0-rc.1 by @lerencao in https://github.com/starcoinorg/starcoin/pull/3084
* Fix db upgrade by @jolestar in https://github.com/starcoinorg/starcoin/pull/3091
* metrics use coarsetime::Instant by @nkysg in https://github.com/starcoinorg/starcoin/pull/3089
* release v1.9.0-rc.2 by @lerencao in https://github.com/starcoinorg/starcoin/pull/3092
* [storage] Fix storage upgrade break by invalid record, just print a warning log by @jolestar in https://github.com/starcoinorg/starcoin/pull/3094
* Fix readiness probe since pipe not working by @sanlee42 in https://github.com/starcoinorg/starcoin/pull/3095
* release v1.9.0 by @lerencao in https://github.com/starcoinorg/starcoin/pull/3097
* [functional-test] test Hash gas used by @jolestar in https://github.com/starcoinorg/starcoin/pull/3033

**Full Changelog**: https://github.com/starcoinorg/starcoin/compare/v1.8.0...v1.9.0

<hr /><em>This discussion was created from the release <a href='https://github.com/starcoinorg/starcoin/releases/tag/v1.9.0'>v1.9.0</a>.</em>

**Comment by nanne007** [comment]
## Starcoin V1.9.0 升级指南

Starcoin v1.9.0 包含了一系列重大更新，其中比较重要的特性包括：

- 虚拟机新增了 几个重要的 native 方法，使得数据跨链成为可能。
- 节点增加了跨链所需要的数据验证接口，底层数据库做了不兼容的更新。
- stdlib 新增了 `U256` 等重要数据结构。

因为以上的特性更新，**本次版本是一个分叉版本，请节点尽快升级到该版本**。
另外，由于数据库做了不兼容的字段更新，该版本在启动时，会自动升级数据库，耗时大概在**一个小时**左右，这期间，节点无法对外提供服务。
强烈建议，在启动新版本之前，将数据目录做个备份。

以下给出示例的操作命令：

``` bash
# 进入 starcoin 的数据目录（这里以默认的目录为实例）
cd ~/.starcoin/
# 备份 main 节点的数据。
cp -r main main.bak

# 启动最新版 v1.9.0 的 starcoin
starcoin -n main

# 启动后，节点会自动升级数据库。

```

如有疑问，请评论区讨论。

**Comment by nanne007** [comment]
## Starcoin v1.9.0 upgrade guide

Starcoin v1.9.0 contains many features and updates for starcoin bridge.
It's not compatible with previous versions, and it will cause fork if you don't upgrade timely.
So PAY ATTENTION to this release.

**Besides, this release will do important ugprades to underlying databases,
so backup your database before do the upgrading.
It will need about ~1hour before it can provide service for outside world.**

The following is an example of how to do it:


``` bash
# enter data dir of starcion
cd ~/.starcoin/
# backup data of main network.
cp -r main main.bak

# satrt starcoin  v1.9.0.
starcoin -n main

# wait starcoin finish its db-upgrade work.
```

**Comment by nanne007** [comment]
Also, the fork is expected to happen at  Dec 31 (next Friday), so please upgrade before this time.

**Comment by jolestar** [comment]
### 接口兼容性说明：

1. TransactionInfo 增加新字段 `transaction_global_index`，影响接口 `chain.get_transaction_info`
2. TransactionEventResponse 增加新字段  event_index，影响接口 `chain.get_events_by_txn_hash`
3. 新增接口 chain.get_transaction_proof, 提供交易以及事件证明
4. 接口输出中的 BlockInfo 字段变更, 影响接口 `chain.info`,`chain.get_block_txn_infos`
    * `block_id` 改为 `block_hash`, 对外接口中统一用 `block_hash` 替代 `block_id`.
    * txn_accumulator_info 和 block_accumulator_info 中的 `num_leaves` 和 `num_nodes` 字段从 u64 变为 string, 避免越界

### JSON RPC compatibility notes:

1. TransactionInfo add new field `transaction_global_index`, affecting the `chain.get_transaction_info` API.
2. TransactionEventResponse add new field `event_index`, affecting the `chain.get_events_by_txn_hash` API.
3. Add an API chain.get_transaction_proof, provider the transaction and event proof.
4. BlockInfo field change in the API output, affecting the `chain.info`,`chain.get_block_txn_infos` API.
    * `block_id` is changed to block_hash, and block_hash is used instead of block_id in the external interface.
    * The `num_leaves` and `num_nodes` fields in txn_accumulator_info and block_accumulator_info are changed from u64 to string to avoid out-of-bounds
